lib/request_handler/fieldsets_parser.rb

```diff
@@ -7,7 +7,8 @@ class FieldsetsParser
     def initialize(params:, allowed: {}, required: [])
       @params = params
       allowed.each_value do |option|
-        raise InternalArgumentError, allowed: 'must be a Enum' unless option.is_a?(Dry::Types::Enum)
+        raise InternalArgumentError, allowed: 'must be a Enum or a Boolean' unless
+              option.is_a?(Dry::Types::Enum) || option.is_a?(TrueClass) || option.is_a?(FalseClass)
       end
       @allowed = allowed
       raise InternalArgumentError, allowed: 'must be an Array' unless required.is_a?(Array)
```

Comment:
not your change but can you please fix this error message. I think this should be
```
raise InternalArgumentError, required: 'must be an Array' unless required.is_a?(Array)
```